src/overlay005/field_control.c

```diff
@@ -0,0 +1,1087 @@
+
+            sub_02056BDC(fieldSystem, nextMap.unk_00, nextMap.unk_04, 0, 0, v6, 1);
+
+            return TRUE;
+        }
+    }
+
+    Field_PlayerPos(fieldSystem, &playerX, &playerZ);
+
+    tileBehavior = sub_02054F94(fieldSystem, playerX, playerZ);
+
+    if (sub_0205DAF8(tileBehavior) || sub_0205DB28(tileBehavior)) {
+        if (input->transitionDir != DIR_EAST) {
+            return FALSE;
+        }
+    } else if (sub_0205DB04(tileBehavior) || sub_0205DB34(tileBehavior)) {
+        if (input->transitionDir != DIR_WEST) {
+            return FALSE;
+        }
+    } else if (sub_0205DB1C(tileBehavior) || sub_0205DB4C(tileBehavior)) {
+        if (input->transitionDir != DIR_SOUTH) {
+            return FALSE;
+        }
+    } else if (sub_0205DC44(tileBehavior)) {
+        if (input->transitionDir != DIR_EAST) {
+            return FALSE;
+        }
+    } else if (sub_0205DC50(tileBehavior)) {
+        if (input->transitionDir != DIR_WEST) {
+            return FALSE;
+        }
+    }
+
+    if (Field_MapConnection(fieldSystem, playerX, playerZ, &nextMap) == FALSE) {
+        return FALSE;
+    }
+
+    int transitionType;
+
+    if (sub_0205DAEC(tileBehavior)) {
+        transitionType = 1;
+    } else if (sub_0205DC44(tileBehavior)) {
+        transitionType = 3;
+    } else if (sub_0205DC50(tileBehavior)) {
+        transitionType = 3;
+    } else if (sub_0205DAF8(tileBehavior) || sub_0205DB28(tileBehavior)
+            || sub_0205DB04(tileBehavior) || sub_0205DB34(tileBehavior)
+            || sub_0205DB1C(tileBehavior) || sub_0205DB4C(tileBehavior)) {
+        sub_02056C18(fieldSystem, nextMap.unk_00, nextMap.unk_04, 0, 0, input->transitionDir);
+        return TRUE;
+    } else {
+        return FALSE;
+    }
+
+    // these statements are unreachable, but required for matching
+    sub_02056BDC(fieldSystem, nextMap.unk_00, nextMap.unk_04, 0, 0, input->transitionDir, transitionType);
+
+    return TRUE;
+}
+
+u16 Field_TileBehaviorToScript (FieldSystem *fieldSystem, u8 behavior)
+{
+    int playerDir = PlayerAvatar_GetDir(fieldSystem->playerAvatar);
+
+    if (sub_0205DBE4(behavior) && playerDir == DIR_NORTH) {
+        return 2018;
+    } else if (sub_0205DC80(behavior)) {
+        return 2500;
+    } else if (sub_0205DC8C(behavior)) {
+        return 2501;
+    } else if (sub_0205DC98(behavior)) {
+        return 2502;
+    } else if (sub_0205DCA4(behavior)) {
+        return 2503;
+    } else if (sub_0205DCB0(behavior)) {
+        return 2504;
+    } else if (sub_0205DCBC(behavior)) {
+        return 2505;
+    } else if (sub_0205DCC8(behavior)) {
+        return 2506;
+    } else if (sub_0205DCD4(behavior)) {
+        return 2507;
+    } else if (sub_0205DDB4(behavior)) {
+        return 10006;
+    } else if (sub_0205DBF0(behavior)) {
+        return 2508;
+    } else if (sub_0205DDC0(behavior)) {
+        return 2030;
+    } else if (sub_0205DE84(behavior) && playerDir == DIR_NORTH) {
+        return 10100;
+    }
```

Comment:
Just noting that I've looked at some of these before (mostly the 2xxx scripts). If you want to start looking into those, here's a table from the binary community that can help you get started on them. You could view these files using DSPRE, maybe, and use that to ascertain behavior.

Starting to document some constants for these commonscripts would be a huge help in eventually getting field scripts documented (and making the repo itself much more moddable).

![image](https://github.com/pret/pokeplatinum/assets/118140365/3c4887a4-0616-45c0-9f1a-429c138d6120)

Reply:
(To clarify, I'm not suggesting you do this as part of *this* PR).